src/userAgent.js

```diff
@@ -54,16 +56,31 @@ var UserAgent = Class(function () {
 
   if (isAndroid) {
     osType = 'Android';
-    osVersionString = ua.match(/Android[/\s][\d.]+/)[0];
-    osVersion = osVersionString.match(/[\d.]+/)[0];
+    match = ua.match(/Android[/\s][\d.]+/);
+    if (match) {
+      osVersionString = match[0];
+      osVersion = osVersionString.match(/[\d.]+/)[0];
+    } else  {
+      osVersionString = osVersion = '';
+    }
   } else if (isIPhoneOS) {
     osType = 'iPhone OS';
-    osVersionString = ua.match(/iPhone OS [0-9_]+/)[0];
-    osVersion = osVersionString.match(/[0-9_]+/)[0].replace(/_/g, '.');
+    match = ua.match(/iPhone OS [0-9_]+/);
+    if (match) {
+      osVersionString = match[0];
+      osVersion = osVersionString.match(/[0-9_]+/)[0].replace(/_/g, '.');
```

Comment:
Same comment as above, we should protect against the RegEx failing to match here